File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        
        for(int i = 3; i < brown+yellow; i++){
            int w = (brown+yellow) / i;
            if(w>=i){
                if((i-2)*(w-2)==yellow){
                    answer[0]=w;
                    answer[1]=i;
                    break;
                }
            }
        }
        
        
        return answer;
    }
}

File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = []
    yx = 0
    yy = 0
    
    for i in range(1, int(yellow**0.5)+1):
        if yellow % i == 0:
            yx = yellow // i
            yy = i
            if 2*yx + 2*yy + 4 == brown:
                answer.append(yx+2)
                answer.append(yy+2)
                break
    
    answer.sort(reverse = True)
            
    return answer